widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogManager.java

```diff
@@ -28,12 +28,23 @@ public void offer(DialogState state) {
         callback.emitDialog(state);
     }
 
+    public void add(DialogState state) {
+        boolean isDialogShown = isDialogShown();
+
+        stateDeque.removeFirstOccurrence(state);
+        stateDeque.offer(state);
+
+        if (!isDialogShown) {
+            callback.emitDialog(state);
+        }
+    }
+
     private DialogState remove() {
-        return stateDeque.pollLast();
+        return stateDeque.pollFirst();
     }
 
     private DialogState getCurrent() {
-        return stateDeque.peekLast();
+        return stateDeque.peekFirst();
```

Comment:
Guys, maybe I made a mistake on it. Need to write tests for it.

Reply:
The previous implementation was working with the `LIFO` method, you changed it to `FIFO`.
I thought that new dialogs/events/warnings are more important than already opened ones.

Reply:
I did it cause the Deque can add the element only on the end(or start) of the stack. To use LIFO in our add method we should add an element to the penultimate place.
So, to use LIFO we can:
1. Use hack
```
public void add(DialogState state) {
        ...
        DialogState currentState = remove();
        stateDeque.offer(state);
        stateDeque.offer(currentState);
        ...
    }
```
2. Use List instead of Deque

Reply:
Agree with you, but I'm not sure that we need to keep `OverlayDialog` or `ExitQueueDialog` in case of `NoMoreOperatorsAvailable` received because after that there is no need for `OverlayDialog` or `ExitQueueDialog` and we must close the chat screen anyways. In this case no need for the `add` method anymore.

Reply:
Maybe you are right about ExitQueueDialog. But OverlayDialog we can show only once. After a user interacts with NoMoreOperatorsAvailable it disappears.
Anyway, I returned to the previous logic of displaying dialogs, may be it made sence according to the requirements.

Reply:
That was only my thoughts I really don't know about the requirements, in this case maybe it makes sense to remove only `ExitQueueDialog` and keep the `add` method as it was.

Reply:
@DavDo but this is your changes regarding `ExitQueueDialog`
https://github.com/salemove/android-sdk-widgets/pull/338/files#diff-23ee758ebee1fe03bc71ea9297219470e1fb17c242291fa8f016be1e926d939dR137

Reply:
I'm sorry I forgot about that bug.